### ISSUE TITLE
PHP Fatal error:  Uncaught Exception: Could not read nfdump profile directory

### ISSUE THREAD

**Issue by klausagnoletti** [mbolli/nfsen-ng#110]
Hi
I just did a new install of nfsen-ng and I have a couple of problems but can't find the cause.

Here's the full error I get when running cli.php to import existing data:

```
www-data@librenms:~/html/nfsen-ng/backend$ ./cli.php -f -p -ps import
2024-08-16 08:38:18 CLI: Starting import
Resetting existing data...


Processing 1 sources...                                                                                             0.0% 0/1097 ETC: ???. Elapsed: < 1 sec [>                              ]PHP Fatal error:  Uncaught Exception: Could not read nfdump profile directory /var/nfdump/profiles-data/live in /var/www/html/nfsen-ng/backend/common/Import.php:48
Stack trace:
#0 /var/www/html/nfsen-ng/backend/cli.php(73): mbolli\nfsen_ng\common\Import->start()
#1 {main}
  thrown in /var/www/html/nfsen-ng/backend/common/Import.php on line 48
```

rrd files are created but show no data (as one would expect).

I've been trawling old issues and I am pretty sure that permissions are correct:
```
srkag@librenms:/var/www/html/nfsen-ng/backend/datasources/data$ ls -la /var/nfdump/profiles_data/live/
total 12
drwxr-xr-x 3 root root 4096 Aug 15 12:12 .
drwxr-xr-x 3 root root 4096 Aug 15 12:06 ..
drwxr-xr-x 3 root root 4096 Aug 16 08:45 isr1100
```

I am able to dump data as the `www-data` user when running `cli.php` manually:
```
www-data@librenms:~/html/nfsen-ng/backend$ nfdump -I -r /var/nfdump/profiles_data/live/isr1100/2024/08/16/nfcapd.202408160755
Ident: isr1100
Flows: 4934
Flows_tcp: 3592
Flows_udp: 1326
Flows_icmp: 16
Flows_other: 0
Packets: 274857
Packets_tcp: 85069
Packets_udp: 189758
Packets_icmp: 30
Packets_other: 0
Bytes: 216519155
Bytes_tcp: 67823312
Bytes_udp: 148691749
Bytes_icmp: 4094
Bytes_other: 0
First: 1723794900
Last: 1723795200
msec_first: 0
msec_last: 0
```

I am also pretty sure that the data structure is correct:
```
srkag@librenms:~$ tree -h /var/nfdump/profiles_data/live/isr1100/
[4.0K]  /var/nfdump/profiles_data/live/isr1100/
├── [4.0K]  2024
│   └── [4.0K]  08
│       └── [4.0K]  16
│           ├── [ 12K]  nfcapd.202408160000
│           ├── [ 12K]  nfcapd.202408160005
│           ├── [ 11K]  nfcapd.202408160010
│           ├── [ 12K]  nfcapd.202408160015
```

Here's how I run nfcapd:
`root     2793066  0.0  0.0  24536  2556 ?        S    08:08   0:00 /usr/bin/nfcapd -D -P /run/nfcapd.default.pid -z -S 1 -T all -l /var/nfdump/profiles_data/live/isr1100/ -p 2055`

nfsen-ng runs as a service (directly copied from installation docs):
```
[Unit]
Description=nfsen-ng
After=network-online.target

[Service]
Type=simple
RemainAfterExit=yes
restart=always
startLimitIntervalSec=0
restartSec=2
ExecStart=su - www-data --shell=/bin/bash -c '/var/www/html/nfsen-ng/backend/cli.php start'
ExecStop=su - www-data --shell=/bin/bash -c '/var/www/html/nfsen-ng/backend/cli.php stop'

[Install]
WantedBy=multi-user.target

```
php stuff is installed:
```
ii  libapache2-mod-php8.3                  8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        server-side, HTML-embedded scripting language (Apache 2 module)
ii  php-common                             2:94+ubuntu22.04.1+deb.sury.org+2            all          Common files for PHP packages
ii  php-pear                               1:1.10.12+submodules+notgz+20210212-1ubuntu3 all          PEAR Base System
ii  php-xml                                2:8.3+94+ubuntu22.04.1+deb.sury.org+2        all          DOM, SimpleXML, WDDX, XML, and XSL module for PHP [default]
ii  php8.3                                 8.3.10-1+ubuntu22.04.1+deb.sury.org+1        all          server-side, HTML-embedded scripting language (metapackage)
ii  php8.3-cli                             8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        command-line interpreter for the PHP scripting language
ii  php8.3-common                          8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        documentation, examples and common module for PHP
ii  php8.3-curl                            8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        CURL module for PHP
ii  php8.3-dev                             8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        Files for PHP8.3 module development
ii  php8.3-fpm                             8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        server-side, HTML-embedded scripting language (FPM-CGI binary)
ii  php8.3-gd                              8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        GD module for PHP
ii  php8.3-gmp                             8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        GMP module for PHP
ii  php8.3-mbstring                        8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        MBSTRING module for PHP
ii  php8.3-mysql                           8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        MySQL module for PHP
ii  php8.3-opcache                         8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        Zend OpCache module for PHP
ii  php8.3-readline                        8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        readline module for PHP
ii  php8.3-xml                             8.3.10-1+ubuntu22.04.1+deb.sury.org+1        amd64        DOM, SimpleXML, XML, and XSL module for PHP
ii  pkg-php-tools                          1.42build1                                   all          various packaging tools and scripts for PHP packages
```

rrd as well:
```
ii  librrd-dev:amd64                       1.7.2-3ubuntu6                               amd64        time-series data storage and display system (development)
ii  librrd8:amd64                          1.7.2-3ubuntu6                               amd64        time-series data storage and display system (runtime library)
ii  rrdcached                              1.7.2-3ubuntu6                               amd64        data caching daemon for RRDtool
ii  rrdtool                                1.7.2-3ubuntu6                               amd64        time-series data storage and display system (programs)
```

Someone asked about the exact same thing in 2020 but for reasons they didn't follow up:
https://github.com/mbolli/nfsen-ng/issues/11#issuecomment-640159017

So bottom line is: I have no freakin' clue as to why it just doesn't work

Can anyone help?

**Comment by mbolli** [comment]
The only "weird" thing I noticed: `profiles_data` instead of `profiles-data`. What do you have in the settings?

**Comment by klausagnoletti** [comment]
Heureka! I kenw it was something stupid.. sems to be working now. Thanks!

**Comment by mbolli** [comment]
@ryan-liu-yes thanks for the catch, fixed in [fd949e5](https://github.com/mbolli/nfsen-ng/commit/fd949e50942f48aa0b664eddeefdb45f6959a1ec)
